api/repositories/droplet_repository_test.go

```diff
@@ -313,6 +313,88 @@ var _ = Describe("DropletRepository", func() {
 		})
 	})
 
+	Describe("GetDroplets", func() {
+		var (
+			dropletRecords []repositories.DropletRecord
+			//			appGUID        string
+			listErr error
+		)
+
+		BeforeEach(func() {
+			meta.SetStatusCondition(&build.Status.Conditions, metav1.Condition{
+				Type:    "Staging",
+				Status:  metav1.ConditionFalse,
+				Reason:  "kpack",
+				Message: "kpack",
+			})
+			meta.SetStatusCondition(&build.Status.Conditions, metav1.Condition{
+				Type:    "Succeeded",
+				Status:  metav1.ConditionTrue,
+				Reason:  "Unknown",
+				Message: "Unknown",
+			})
+			build.Status.Droplet = &korifiv1alpha1.BuildDropletStatus{
+				Stack: dropletStack,
+				Registry: korifiv1alpha1.Registry{
+					Image: registryImage,
+					ImagePullSecrets: []corev1.LocalObjectReference{
+						{
+							Name: registryImageSecret,
+						},
+					},
+				},
+				ProcessTypes: []korifiv1alpha1.ProcessType{
+					{
+						Type:    "rake",
+						Command: "bundle exec rake",
+					},
+					{
+						Type:    "web",
+						Command: "bundle exec rackup config.ru -p $PORT",
+					},
+				},
+				Ports: []int32{8080, 443},
+			}
+			// Update Build Status based on changes made to local copy
+			Expect(k8sClient.Status().Update(testCtx, build)).To(Succeed())
```

Comment:
We have compe up with our own [utility](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/tools/k8s/patch.go#L33) to patch resources. It uses k8s client's `Patch` instead of `Update` - `Update`s are quite probable to be rejected in a ditributed systems causing annoying flakes.

FWIW, we have two simolar patch utilities: [`Patch`](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/tools/k8s/patch.go#L33) and [`PatchResource`](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/tools/k8s/patch.go#L88). `Patch` would try to patch the `Status` subresource, while `PathResource` works on the main resource only. We use `PatchResource` whenever our modificator function does not touch the status, or to update resources that do not have `Status`, such as `Secret`s.

Here is an [example](https://github.com/cloudfoundry/korifi/blob/d0483c8b89cc8112f731f882a8dfe1485f0b6cb5/api/repositories/droplet_repository_test.go#L106) on how to use `Patch`